op-service/txmgr/queue.go

```diff
@@ -51,17 +51,36 @@ func (q *Queue[T]) Wait() error {
 	return q.group.Wait()
 }
 
+// HandleResponse will wait for the response on the first passed channel,
+// and then forward it on the second passed channel (attaching the id). It returns
+// the response error or the context error if the context is canceled.
+func HandleResponse[T any](ctx context.Context, c chan SendResponse, d chan TxReceipt[T], id T) error {
```

Comment:
tiny nit: probably don't need to export this method

Reply:
Yeah good call. I had been using it in some tests in the proposer package but didn't go that way in the end, so I'll unexport this.

Reply:
[8c06a80](https://github.com/ethereum-optimism/optimism/pull/13120/commits/8c06a80cc705ae376ea0583d2849d72c1489528c)